include/kodlab_mjbots_sdk/attitude.h

```diff
@@ -0,0 +1,334 @@
+   *        and uncertainty.
+   *
+   * @param att_quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   * @param ang_rate_bias_in gyroscope bias about <code>[x, y, z]</code> axes
+   * @param att_uncertainty_in attitude uncertainty quaternion
+   * @param ang_bias_uncertainty_in gyroscope bias uncertainty about
+   *                                <code>[x, y, z]</code> axes
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  Attitude(const Eigen::Quaternion<Scalar> &att_quat_in,
+           const Eigen::Vector3<Scalar> &ang_rate_in,
+           const Eigen::Vector3<Scalar> &accel_in,
+           const Eigen::Vector3<Scalar> &ang_rate_bias_in,
+           const Eigen::Quaternion<Scalar> &att_uncertainty_in,
+           const Eigen::Vector3<Scalar> &ang_bias_uncertainty_in,
+           const Eigen::Quaternion<Scalar> &world_offset_in = {1, 0, 0, 0})
+      : att_quat_raw_(att_quat_in),
+        att_quat_(world_offset_in * att_quat_in),
+        ang_rate_(ang_rate_in),
+        accel_(accel_in),
+        ang_rate_bias_(ang_rate_bias_in),
+        att_uncertainty_(att_uncertainty_in),
+        ang_bias_uncertainty_(ang_bias_uncertainty_in),
+        world_offset_(world_offset_in) {}
+
+  /**
+   * @brief Construct a new \c Attitude object from a
+   *        \c mjbots::pi3hat::Attitude object
+   *
+   * @param pi_att attitude object
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  Attitude(const ::mjbots::pi3hat::Attitude &pi_att,
+           const Eigen::Quaternion<Scalar> &world_offset_in = {1, 0, 0, 0})
+      : att_quat_raw_(pi_att.quat.x,
+                      pi_att.quat.y,
+                      pi_att.quat.z,
+                      pi_att.quat.w),
+        att_quat_(world_offset_in * att_quat_raw_),
+        ang_rate_(pi_att.rate_dps.x, pi_att.rate_dps.y, pi_att.rate_dps.z),
+        accel_(pi_att.accel_mps2.x, pi_att.accel_mps2.y, pi_att.accel_mps2.z),
+        ang_rate_bias_(pi_att.bias_dps.x, pi_att.bias_dps.y, pi_att.bias_dps.z),
+        att_uncertainty_(pi_att.attitude_uncertainty.x,
+                         pi_att.attitude_uncertainty.y,
+                         pi_att.attitude_uncertainty.z,
+                         pi_att.attitude_uncertainty.w),
+        ang_bias_uncertainty_(pi_att.bias_uncertainty_dps.x,
+                              pi_att.bias_uncertainty_dps.y,
+                              pi_att.bias_uncertainty_dps.z),
+        world_offset_(world_offset_in) {}
+
+  /**
+   * @brief Update this \c Attitude object's data
+   *
+   * @param att_quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   */
+  void Update(const Eigen::Quaternion<Scalar> &att_quat_in,
```

Comment:
see above comment

Reply:
ALso see above.

Reply:
Resolved in bdfce1322a036db5ad1a8e1a939c31b6042e2025